build.fsx

```diff
@@ -197,7 +503,119 @@ Target.create "ReleaseDocs" (fun _ ->
     Commit.exec tempDocsDir (sprintf "Update generated documentation for version %s" buildVersion)
     Branches.push tempDocsDir
 )
+*)
+
+let rec getUserInput (prompt : string) =
+    Console.Write $"%s{prompt}: "
+    let line = Console.ReadLine ()
+    if String.IsNullOrWhiteSpace line then
+        Console.WriteLine "Entry must be non-whitespace."
+        getUserInput prompt
+    else
+        line
+
+type GitHubAsset =
+    {
+        BrowserDownloadUrl : Uri
+        Name : string
+        ContentType : string
+        Size : uint64
+    }
+
+type GitHubReleaseResponse =
+    {
+        Name : string
+        PublishedAt : DateTime
+        Assets : GitHubAsset list
+    }
+
+type GitHubReleaseRequest =
+    {
+        [<JsonPropertyName "tag_name">]
+        TagName : string
+        [<JsonPropertyName "target_commitish">]
+        TargetCommitish : string
+        [<JsonPropertyName "name">]
+        Name : string
+        [<JsonPropertyName "body">]
+        Body : string
+        [<JsonPropertyName "draft">]
+        IsDraft : bool
+        [<JsonPropertyName "prerelease">]
+        IsPreRelease : bool
+    }
 
+let release (_ : HaveTested) (_ : HaveGeneratedDocs) =
+    Console.WriteLine "Tagging new version and performing GitHub release... "
+
+    let pat =
+        match Environment.GetEnvironmentVariable "github-pat" with
+        | null ->
+            [
+                "Obtain a GitHub personal access token from https://github.com/settings/tokens?type=beta ."
```

Comment:
Note the lack of the old username/password combo here. GitHub themselves would prefer using a PAT, I believe.

Reply:
That is fine. That never worked anyway...I had to log in manually every time.